src/Raven.Server/ServerWide/ClusterStateMachine.cs

```diff
@@ -3101,6 +3103,33 @@ public async Task WaitForIndexNotification(long index, CancellationToken token)
                         break;
                 }
             }
+
+            if (_tasksDictionary.TryGetValue(index, out var tcs) == false)
+            {
+                // the task has already completed
+                // let's check if we had errors in it
+                foreach (var error in _errors)
+                {
+                    if (error.Index == index)
+                        error.Exception.Throw();// rethrow
+                }
+                return;
+            }
+
+            var task = tcs.Task;
+
+            if (task.IsCompleted)
+                return;
+
+            var result = await Task.WhenAny(task, waitAsync);
+
+            if (result.IsFaulted)
+                throw result.Exception;
```

Comment:
Wrap this in another exception. You are modifying the exception stack trace here.

Reply:
Right... maybe `await result` is a cleaner solution?

Reply:
+1 for `await result`